usr/share/rear/output/default/94_grub2_rescue.sh

```diff
@@ -110,6 +110,9 @@ else
     grub-mkconfig -o $grub_conf
 fi
 
+#Finding UUID of filesystem containing /boot 
+grub_boot_uuid=$(df /boot | awk 'END {print $1}' | xargs blkid -s UUID -o value)
```

Comment:
@gdha. `df /boot | awk 'END {print $1}'` returns the file system on which `/boot` stands : either `/boot` (if it is a file system) or `/` (if it is only a directory).
So, if `/boot` is not mounted, the UUID return by the command will be `/` UUID (and not `/boot`).
`grub-mkconfig` has the same behavior. it uses `grub2-probe --target=device /boot` to get the name of the boot device. If `/boot` is not mounted, it returns `/` device instead and `grub.cfg` generation won't be good.
It is not a good idea to not mount `/boot`, `grub-mkconfig` can be automatically run after some packages upgrade (kernel for example); if `/boot `is not mounted, `grub.cfg` won't be updated.

Anyway, your point is correct and we should may be test if a directory like /boot/grub* exist.
If not, we can exist and avoid to generate a bad `grub.cfg`
```
#Test if /boot is mounted 
if ls -ld /boot/grub* > /dev/null 2>&1 ; then
    grub_boot_uuid=$(df /boot | awk 'END {print $1}' | xargs blkid -s UUID -o value)
    ....
else
    echo "Error: /boot is empty, or may be not mounted..."
    exit 1     
fi
```

Your feedback is welcomed